### PR TITLE
NFC - translating frequency unit day/month/year

### PR DESCRIPTION
**Pull request by samuelsov**
Overview
----------------------------------------
Translating day/month/year frequency unit in confirmation and thank you pages.

### CODE REVIEW

**Comment by samuelsov** [comment]
Probably not a good idea after all. Those values comes from the table civicrm_option_value but the name seems to be used instead of the label.

File: templates/CRM/Contribute/Form/Contribution/ThankYou.tpl [hunk 1]
```diff
@@ -133,29 +133,30 @@
 
         {* Recurring contribution / pledge information *}
         {if $is_recur}
+          {capture assign="ts_frequency_unit"}{ts}{$frequency_unit}{/ts}{/capture}
```
Comment (mlutfy) [inline templates/CRM/Contribute/Form/Contribution/ThankYou.tpl]: Avoid sending variables in `ts`: https://docs.civicrm.org/dev/en/latest/translation/#avoid-variables-inside-strings

**Comment by samuelsov** [comment]
It might be a good option after all :
- less risky to simply translate the value here instead of using label as it could be used somewhere to do special action based on the frequency unit : `if $frequency_unit == 'month') ...`
- if we use the name instead of the label, we could use the plural form of ts()

File: templates/CRM/Contribute/Form/Contribution/ThankYou.tpl (hunk 1 above)
Comment (samuelsov) [reply templates/CRM/Contribute/Form/Contribution/ThankYou.tpl]: @mlutfy Do you think it's better - we might want to have an external tpl for this : 

{capture assign="ts_frequency_unit"}
  {if $frequency_unit eq 'day'}{ts}day{/ts}
  {elseif $frequency_unit eq 'week'}{ts}week{/ts}
  {elseif $frequency_unit eq 'month'}{ts}week{/ts}
  {elseif $frequency_unit eq 'year'}{ts}year{/ts}
  {else}{$frequency_unit}
  {/if}
{/capture}

Comment (samuelsov) [reply templates/CRM/Contribute/Form/Contribution/ThankYou.tpl]: I guess it's better because it can be extracted for transifex translation

**Comment by samuelsov** [comment]
New patch but need testing before creating a PR : https://github.com/samuelsov/civicrm-core/commit/9efb4d9385030266a7ab7a9a042f28bc026e4dc5

**Pull request closed without merging**